dask/dataframe/shuffle.py

```diff
@@ -934,14 +934,9 @@ def shuffle_group(df, cols, stage, k, npartitions, ignore_index, nfinal):
         if nfinal and nfinal != npartitions:
             ind = ind % int(nfinal)
 
-    c = ind.values
     typ = np.min_scalar_type(npartitions * 2)
-
-    c = np.mod(c, npartitions).astype(typ, copy=False)
-    np.floor_divide(c, k**stage, out=c)
-    np.mod(c, k, out=c)
-
-    return group_split_dispatch(df, c, k, ignore_index=ignore_index)
+    ind = (ind % npartitions).astype(typ, copy=False) // k**stage % k
```

Comment:
Can you provide a comment here about what's happening.  I think it's sufficiently complex enough that the next reader would benefit from a brief explanation

Reply:
This is just a reformat of the existing code, I can't remember the details of index calculation. @rjzamora do you remember?

Reply:
This was already a pretty "handwavy" piece of code when I first visited shuffling. Here is my understanding of what we are trying to do:

We start with `c` (which labels the final partition-destination index for each row), and `k` (which specifies the total number of groups that we are splitting each partition into for the current shuffling “stage”). The whole point of this logic is to convert the final output index (elements of `c`) into the output index for the current stage (which is only the same as `c` for a single-stage shuffle).

It turns out that the way to transform the final partition-destination index into the index of the current output group is to (1) take the modulus of `c` by `npartitions` (where `npartitions` is the number of partitions of the input DataFrame collection), and then (2) floor-divide that result by `k` to the `stage` power (meaning we divide by 1 for stage 0 and k for stage 1, etc..), and then (3) take the modulus of that result by `k`.

If you are going to reformat that logic, then it may make sense to add a comment explaining that this is what we are going for?